Mathlib/Analysis/Meromorphic/NormalFormAt.lean

```diff
@@ -135,6 +124,146 @@ theorem AnalyticAt.meromorphicNFAt (hf : AnalyticAt 𝕜 f x) :
     MeromorphicNFAt f x := by
   simp [meromorphicNFAt_iff_analyticAt_or, hf]
 
+/-- Meromorphic functions have normal form outside of a discrete subset in the domain of
+meromorphicity. -/
+theorem MeromorphicOn.meromorphicNFAt_codiscreteWithin [CompleteSpace E] {U : Set 𝕜}
+    (hf : MeromorphicOn f U) :
+    { x | MeromorphicNFAt f x } ∈ Filter.codiscreteWithin U := by
+  filter_upwards [hf.analyticAt_codiscreteWithin] with _ ha
+  exact ha.meromorphicNFAt
+
+private lemma WithTop.map_natCast_eq_zero {n : WithTop ℕ}
+  (hn : WithTop.map (Nat.cast : ℕ → ℤ) n = 0) :
+  n = 0 := by
```

Comment:
Can't you more generally prove that `WithTop.map n = 0 \iff n = 0` for `n : ENat`?

Reply:
I can probably do that … but this is only a private lemma that is only used once. I have therefore reduced the lemma to a minimum. Let me know if you want to expand the lemma, and then I will do that.

Reply:
If you generalise it, then it can become an API lemma and you need 0 private lemma

Reply:
OK, will do. Is there a preferred place to put them?

Reply:
Done as suggested.